lua/java/api/profile_config.lua

```diff
@@ -0,0 +1,135 @@
+local async = require('java-core.utils.async').sync
+local get_error_handler = require('java.handlers.error')
+local class = require('java-core.utils.class')
+
+local current_project_path = vim.fn.expand('%:p:h')
+local config_path = vim.fn.stdpath('data') .. '/nvim-java-profiles.json'
```

Comment:
I don't know If I missed this in the last review. You can get the root of the project from the language server.

```lua
vim.lsp.get_clients({ name = 'jdtls' })[1].root_dir
```

We can also get the client using following function
https://github.com/nvim-java/nvim-java/blob/9860a93f113e34c696bf02fcf3504429d5260ff3/lua/java/utils/jdtls.lua?plain=1#L5-L15

Reply:
@atm1020 With this, we have to wait for jdtls to start first though. What do you think?

Reply:
@s1n7ax Yes, the initialization can be a bit tricky,  however as you mentioned, autocmd can be help on this.

In the other hand i think it might be better to keep the profile editing independent of the jdtls init, bacuse the profiles editing is depending one point on the jdtls/dap init, when you edit and save a profile config (the dap update runs after the profile is saved successfully, and there is a little bug here: https://github.com/nvim-java/nvim-java/pull/146#issuecomment-2032461010) . 
I think it can be resolved easily, if we check  the jdtls client is initialized before saving the profile as i mention before.
So this can work independently with minimal dev effort.

Reply:
@atm1020 Yeah but the issue is path is relative to the current buffer which could be anything. Not even a java file.

![image](https://github.com/nvim-java/nvim-java/assets/18459807/b17420e5-1bd0-4417-ad5a-9fd561010f29)

I guess you can use the same function I have used here. Just directly call it and get the path

https://github.com/nvim-java/nvim-java-core/blob/308c17c732eb29f5332b1f5e0e347206591f8660/lua/java-core/ls/servers/jdtls/init.lua?plain=1#L85

Reply:
@s1n7ax Unfortunately, I forgot that the multimodule application is an option. Currently, there's only one Profile selector per root directory (thus, only one profile selector for all the modules). I believe we can resolve this issue by using the main class selector (like in the runner api) and we can use this info the JSON config as well (the selected mainclass) . But for this, we have to use the jdtls client. If we try to set up this way, we can use the jdtl client to get the project root directory as you suggested. What do you think ?

Reply:
- @s1n7ax  I have implemented these changes, and I think, now works well for both simple and multi-module projects.
- Before you test, make sure to delete the profile config `~/.local/share/nvim/nvim-java-profiles.json`. (because of the new config format) 

  new config format:
  ```json 
  {
    "roo_dir" : {
 	  "mod1 -> main_class" : [
 		  {
 			  "vm_args": "-Xmx1024m",
 			  "prog_args": "arg1 arg2",
 			  "name": "profile_name1",
 			  "is_active": true
 		  },
 		  {
 			  "vm_args": "-Xmx1024m",
 			  "prog_args": "arg1 arg2",
 			  "name": "profile_name2",
 			  "is_active": false
 		  }
 	  ],
 	  "mod2 -> main_class" : [  ]
     }
  }
  
  ```
- I ran some tests, and I found that when you open a multi module project, jdtls returns the root directory of the first opened module, not the folder where the module directories are located. So I think the `vim.fn.getcwd()` is the best  choice for this (currently i used this one) .